### PR TITLE
Add systemd init services

### LINKED ISSUE #123: Ansible didn't continue after enable service step
This line https://github.com/DavidWittman/ansible-redis/blob/master/tasks/server.yml#L24
```
fatal: [104.199.185.189]: FAILED! => {"changed": false, "failed": true, "msg": "Could not find the requested service \"'redis_6379'\": "}
```

I have to manually SSH into the server and run below command
`systemctl enable redis_6379`

I use Debian Jessie on GCP

The playbook then stop again when i'm trying to deploy sentinel.

I already set `ansible_become=true` in the host file
i use ansible 2.2 on OSX.

Please advise

### PR DESCRIPTION
**Pull request by danielkza**
Fixes #123, #121, and #91 (also #79 and  #75 for systemd systems).

### CODE REVIEW

**Comment by DavidWittman** [comment]
Thanks @danielkza, these changes look great. Running some tests against them now.

**Comment by danielkza** [comment]
I forgot to mention it, but the unit files are based on the ones currently in Debian Sid.

File: tasks/server.yml [hunk 1]
```diff
@@ -19,12 +19,26 @@
...
+      - default/redis.service.j2
+      paths:
+        - ../templates
+  when: redis_as_service and ansible_service_mgr | default == 'systemd'
 
 - name: set redis to start at boot
   service:
     name: "{{ redis_service_name }}"
     enabled: yes
+    daemon_reload: "{{ True if ansible_service_mgr | default == 'systemd' else omit }}"
```
Comment (DavidWittman) [inline tasks/server.yml]: Does `daemon_reload` work in the `service` module? I only see it referenced in the documentation for the `systemd` module.

**Comment by danielkza** [comment]
The service module forwards arguments to the implementation module. Ansible
2.2 knows not to forward arguments that the implementation can't handle,
but earlier versions require the conditional.

Em ter, 15 de nov de 2016 12:58, David Wittman <notifications@github.com>
escreveu:

**Comment by DavidWittman** [comment]
@danielkza this is good to merge after you address [my comment](https://github.com/DavidWittman/ansible-redis/pull/124#discussion_r88032729).

File: tasks/server.yml (hunk 1 above)
Comment (danielkza) [inline tasks/server.yml]: The service module forwards arguments to the implementation module AFAIK. Ansible
2.2 knows not to forward those that the implementation can't handle,
but earlier versions require the conditional.

**Comment by DavidWittman** [comment]
Heh... we updated at the same time. Awesome. Is there documentation on that behavior anywhere?

**Comment by DavidWittman** [comment]
Tagged as 1.2.0

**Comment by danielkza** [comment]
@DavidWittman I noticed it by looking at the output of the service module - it includes a `daemon_reload` parameter when systemd is the init system. You might have a point that it isn't good to rely on it. Triggering a daemon-reload manually should also work just fine.

Now that you've already merged it shouldn't matter too much. I might have one or two more PRs for you today, so if you want to hold a bit before releasing it might save you some work :)

**Comment by DavidWittman** [comment]
Meh... I already tagged 1.2.0, and it might be a while before I get to testing the other changes. Thanks again.

**Comment by danielkza** [comment]
@DavidWittman Now that I think about, you have a valid point: there's no guarantee that the ar

**Pull request merged**